server/src/test/java/org/elasticsearch/action/search/DfsQueryPhaseTests.java

```diff
@@ -278,7 +290,7 @@ public void sendExecuteQuery(
                         queryResult.decRef();
                     }
                 } else if (request.contextId().getId() == 2) {
-                    throw new UncheckedIOException(new MockDirectoryWrapper.FakeIOException());
+                    listener.onFailure(new UncheckedIOException(new MockDirectoryWrapper.FakeIOException()));
```

Comment:
This test made no sense IMO, we were reproducing a hard bug here. We can't have exceptions bubble up here, that would breaking the counting response listener and is not something that happens in production code -> adjusted the test to behave like the production code would and pass the exception to the listener.